lib/spack/spack/cmd/spec.py

```diff
@@ -66,12 +70,15 @@ def spec(parser, args):
               'show_types': args.types,
               'install_status': args.install_status}
 
+    if not args.specs:
+        tty.die("spack spec requires at least one spec")
+
     for spec in spack.cmd.parse_specs(args.specs):
         # With -y, just print YAML to output.
         if args.yaml:
             if spec.name in spack.repo.path or spec.virtual:
                 spec.concretize()
-            print(spec.to_yaml())
+            sys.stdout.write(spec.to_yaml())
```

Comment:
could you add a comment why you changed this? When I'm just reading the code I'd be: Why is this not just a print?

Independent of that you could also do
```
from __future__ import print_function
print(spec.to_yaml(), end='')
```
(albeit I don't know whether this also works with python 2.6

Reply:
Done.  It'll work in python 2.6 if you `from __future__ import print_function`.  But I think `sys.stdout.write` is clearer if you just want to do normal file I/O on `stdout`